THRIFT-5418: C++ to_string and ostream operator not always generated

dsandbrink opened the pull request:
If a thrift file only contains an enum, then the c++ source file with
the implementation for to_string and ostream operator<< is not
generated.
This bug was introduced in PR #2095 / THRIFT-5168.

<!-- Explain the changes in the pull request below: -->
  
To fix this issue also set `has_members_` to true in `t_cpp_generator::generate_enum`, not only in `t_cpp_generator::generate_cpp_struct`.

<!-- We recommend you review the checklist/tips before submitting a pull request. -->

- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?  (not required for trivial changes)
- [x] If a ticket exists: Does your pull request title follow the pattern "THRIFT-NNNN: describe my issue"?
- [x] Did you squash your changes to a single commit?  (not required, but preferred)
- [x] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [ ] If your change does not involve any code, include `[skip ci]` anywhere in the commit message to free up build resources.

<!--
  The Contributing Guide at:
  https://github.com/apache/thrift/blob/master/CONTRIBUTING.md
  has more details and tips for committing properly.
-->

emmenlau reviewed (approved):
Looks good. I tried to identify other cases that may need `has_members_ = true;` but nothing else jumps to the eye. So currently only `generate_cpp_struct()` and `generate_enum()` qualify as having members. Is that reasonable?

emmenlau commented:
@dsandbrink could you kindly rebase on latest master and do a forced push of your PR? I've fixed the build and it should now mostly work. The errors before are most likely unrelated to your change, but I'd like to see more green lights if possible...

dsandbrink commented:
> @dsandbrink could you kindly rebase on latest master and do a forced push of your PR? I've fixed the build and it should now mostly work. The errors before are most likely unrelated to your change, but I'd like to see more green lights if possible...

Sure, I updated both PRs.

emmenlau commented:
Thanks for the nice PR. Merged.
